pkg/3scale/amp/component/s3.go

```diff
@@ -14,6 +14,11 @@ const (
 	S3SecretAWSSecretAccessKeyFieldName = "AWS_SECRET_ACCESS_KEY"
 )
 
+const (
+	StorageServiceEndpointScheme = "STORAGE_SERVICE_ENDPOINT_SCHEME"
```

Comment:
System will stick to the convention from other AWS S3 config fields and name these new ones as well `AWS_X`. See https://github.com/3scale/porta/pull/1522.

From System (3scale) perspective, it's AWS S3 API. These values are used to establish a connection to AWS S3 (or a compatible service that follows AWS S3 API). The fields are named after the API, which in this case is a AWS (S3) API.

```suggestion
	StorageServiceEndpointScheme = "AWS_HOSTNAME"
```

Reply:
I don't really like the idea of using AWS_* for the environment variables.
Precisely the motivation of this PR is to allow S3 api-compatible services to be used, and those services might not be used in AWS at all.
I think it is confusing for a user seeing AWS_PROTOCOL or AWS_HOSTNAME when they want to configure a S3 api-compatible service (like min.io) that is for example deployed in Azure...
There could be an intermediate solution which is envvars being named AWS_* as you proposed but the config exposed to the user (in the ConfigMap, in the CR fields, in the OpenShift templates) named another way that does not include the AWS prefix...

Reply:
Also AWS_PROTOCOL by itself even if only used in AWS (which is not the case anymore) I don't think it's a good name. There are multiple services in AWS, so AWS_PROTOCOL what does it mean? what protocol? the protocol used in s3? in rds? in an EC2 machine? it's confusing too

Reply:
I understand your concern, @miguelsorianod. However, @3scale/system will stick to the current convention for the time being.

The `AWS_` prefix clearly states the nature of the connection to be established. It's about the API, not about the service provider behind. (In fact, it should even be called `AWS_S3_` IMHO, to make them ever more clear.)

Of course, we cannot make any assumptions about the service provider behind. There might be a proxy, for example, routing requests that we believe to be going to AWS S3, in reality, to a different provider (even when the value of the "hostname" config option says `aws.amazon.com`). From our perspective however, we implemented a connection to an AWS S3-provided service; System uses a library that trusts it is really AWS S3 providing the service (written and maintained by Amazon!) For now, anything other than AWS S3 can still be used, but the risk is on the customer. If we update the library and integration with a pretend-to-be-AWS service breaks, we cannot test that, we cannot track that, we can barely explain that to the customer.

And we have analogous cases, such as `THINKING_SPHINX_*`, `REDIS_*`, `MEMCACHE_*`. Those are all configuration options related to specific products in the market (free or commercial) that we once decided to commit to. One could argue now that we are actually committed to those products' APIs, rather than to the products providers themselves. Yet, we won't rename our env vars, for example, because some player in the market decided to create an alternative to Redis that turns out to copy Redis API and promised to be 100% compatible.

Integration with AWS S3 (or AWS S3 compatible services, if you prefer) is not being introduced with this PR. It's already there, and all other env vars related are names `AWS_*`.

Reply:
Everybody uses `AWS_*` environment variables, I do not see why we don't do that.
Min.io uses it in its client

https://github.com/minio/minio/blob/46b4dd8e20dbcab872206ff7b90774f243ed5bfd/docs/lifecycle/README.md#2-enable-bucket-lifecycle-configuration